### PR TITLE
Build in obs

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

This PR adds submission of wanda to OBS.

More in detail
- a hack directory added with necessary scripts for CI
- added suse Docker packaging files
- vendored rhai rustler cargo deps

Reminder to self: before merging update obs meta and merge web bci base update https://github.com/trento-project/web/pull/1130

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
 
   tlint:
     name: Lint checks
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (fabriziosestito) [inline .github/workflows/ci.yaml:59]: why is this needed?

Comment (nelsonkopliku) [reply .github/workflows/ci.yaml:59]: No strict need detected actually, I guess I just pinned os version for all the jobs.
I'll revert that to `ubuntu-latest` if it is preferable.

Comment (fabriziosestito) [reply .github/workflows/ci.yaml:59]: If you don't need to do so, I suggest dropping the commit.
This might be needed if we decide to go with a matrix testing and test against an OTP version that is not supported by ubuntu 22.04, we can re-add modification in that PR if needed (there is already a debt for this).

Comment (nelsonkopliku) [reply .github/workflows/ci.yaml]: Makes sense, thanks! Changed back to latest

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -91,7 +99,7 @@ jobs:
             _build/test
             priv/plts
           key: ${{ runner.os }}-${{ env.OTP_VERSION }}-${{ env.ELIXIR_VERSION }}-${{ hashFiles('mix.lock') }}
-      - name: Check for Unused Dependencies
+      - name: Check for unused dependencies
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:102]: I don't really get this name change, as all the remaining step names are in title case

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -221,3 +233,77 @@ jobs:
...
+    name: Commit the project on OBS
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' || (github.event_name == 'push' && github.ref_name == 'main') || github.event_name == 'workflow_dispatch'
+    needs: [static-code-analysis, test]
+    container:
+      image: ghcr.io/trento-project/continuous-delivery:main
+      env:
+        GITHUB_OAUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        DEST_FOLDER: "/tmp/osc_project"
+        NAME: trento-wanda-image
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:247]: This naming could be confusing, as the container in the public environment is named `wanda`, without any trento prefix.
We should think carefully how we want to name this

File: packaging/suse/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: Apache-2.0
+#!BuildTag: trento/trento-wanda:latest
+#!BuildTag: trento/trento-wanda:%%VERSION%%
+#!BuildTag: trento/trento-wanda:%%VERSION%%-build%RELEASE%
+#!UseOBSRepositories
+
+FROM bci/rust:1.66 AS release
```
Comment (arbulu89) [inline packaging/suse/Dockerfile:7]: Let's test this first in IBS before merging

Comment (arbulu89) [inline packaging/suse/Dockerfile:7]: Just out of curiosity, why do we need a `rust` image here if the rust files are already vendored?

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (nelsonkopliku) [reply .github/workflows/ci.yaml:102]: Just copied from web :smile: Putting back as it was.

File: packaging/suse/Dockerfile (hunk 4 above)
Comment (nelsonkopliku) [reply packaging/suse/Dockerfile:7]: Tried and seems to be working https://build.opensuse.org/package/show/home:nkopliku:trento-factory/trento-wanda-image

Comment (nelsonkopliku) [reply packaging/suse/Dockerfile:7]: Because the deps are vendored ineed, not built. They get built later in OBS.

Comment (arbulu89) [reply packaging/suse/Dockerfile:7]: That's an OBS url. Not IBS, which has a different repositories/package configuration

Comment (nelsonkopliku) [reply packaging/suse/Dockerfile:7]: True :sweat_smile: I sent you the link to IBS

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (nelsonkopliku) [reply .github/workflows/ci.yaml:247]: I see. It just follows the same naming pattern we use for web.
| Project | gh image name | OBS/IBS image name |
|---	|---	|---	|
| web | `trento-web` | `trento-web-image` |
| wanda | `wanda` | `trento-wanda-image` |

We are somewhat consistently inconsistent :sweat_smile: 
What should we do?

Comment (nelsonkopliku) [reply .github/workflows/ci.yaml:247]: Options:
- rename the gh image to `trento-wanda`.
- move away from `wanda` to something checks engine related (`trento-checks`, `trento-checks-engine` or similar)

I'd go for the first one, sounds simpler.

Comment (fabriziosestito) [reply .github/workflows/ci.yaml:247]: +1 to keep consitency with the trento prefix if possible

Comment (nelsonkopliku) [reply .github/workflows/ci.yaml:247]: https://github.com/trento-project/wanda/pull/173/commits/8a1590b339ad85aaee969d809d395871535e8d85

**Review by fabriziosestito — APPROVED** [review]

**Review by arbulu89 — APPROVED** [review]
